main.go

```diff
@@ -0,0 +1,422 @@
+	(*d).DBClusterIdentifier = *databaseInstances.DBInstances[0].DBClusterIdentifier
+
+	databaseClusters, err := client.DescribeDBClusters(&rds.DescribeDBClustersInput{DBClusterIdentifier: &d.DBClusterIdentifier})
+	if err != nil {
+		return errors.Wrapf(err, "unable to describe the DB Cluster")
+	}
+
+	if len(databaseClusters.DBClusters) == 0 {
+		return errors.Wrapf(err, "list of DB Clusters empty")
+	}
+	for _, member := range databaseClusters.DBClusters[0].DBClusterMembers {
+		if *member.DBInstanceIdentifier == d.DBInstanceIdentifier {
+			(*d).IsClusterWriter = *member.IsClusterWriter
+		}
+	}
+	return nil
+}
+
+func (d *DBInstance) getNewClassType() (string, error) {
+	newClass, err := d.increaseSize()
+	if err != nil {
+		return "", err
+	}
+	log.Infof("New DB instance class (%s)", newClass)
+	return newClass, nil
+}
+
+func (d *DBInstance) getSetDBInstanceClass() bool {
+	for i, dbClass := range DBInstanceClasses {
+		if d.DBInstanceClass == dbClass {
+			(*d).SizeIndex = i
+			return true
+		}
+	}
+	return false
+}
+
+func (d DBInstance) increaseSize() (string, error) {
+	newIndex := d.SizeIndex + 1
+	if (newIndex + 1) >= len(DBInstanceClasses) {
+		return "", errors.Errorf("Maximum instance size used. Index out of range")
+	}
+	return DBInstanceClasses[newIndex], nil
+}
+
+func (d *DBInstance) changeDatabaseClass(client *rds.RDS, dbInstanceClass string) error {
+	modifyDBInstanceInput := &rds.ModifyDBInstanceInput{
+		ApplyImmediately:     aws.Bool(true),
+		DBInstanceClass:      aws.String(dbInstanceClass),
+		DBInstanceIdentifier: aws.String(d.DBInstanceIdentifier),
+	}
+	log.Infof("Upgrading database (%s) to class (%s)", d.DBInstanceIdentifier, dbInstanceClass)
+	_, err := client.ModifyDBInstance(modifyDBInstanceInput)
+	if err != nil {
+		return errors.Wrapf(err, "unable to upgrade database to new class")
+	}
+	wait := 300
+	log.Infof("Waiting up to %d seconds for db instance to start modifications...", wait)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(wait)*time.Second)
+	defer cancel()
+	err = d.waitForDBInstanceStartModifications(ctx, client)
+	if err != nil {
+		return err
+	}
+
+	wait = 1000
+	log.Infof("Waiting up to %d seconds for db instance to become available...", wait)
+	ctx, cancel = context.WithTimeout(context.Background(), time.Duration(wait)*time.Second)
+	defer cancel()
+	err = d.waitForDBInstanceReady(ctx, client)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *DBInstance) waitForDBInstanceReady(ctx context.Context, client *rds.RDS) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return errors.Wrap(ctx.Err(), "timed out waiting for database to get ready")
+		default:
+			var shouldWait bool
+
+			databaseInstances, err := client.DescribeDBInstances(&rds.DescribeDBInstancesInput{DBInstanceIdentifier: &d.DBInstanceIdentifier})
+			if err != nil {
+				return errors.Wrapf(err, "unable to describe DB instance")
```

Comment:
I think you may want to log the error, but not return here. Same with some of the checks below. We should probably just sleep and try again instead of failing the whole process.

Reply:
Yeap you are right on that. Thanks for catching this :)